### PR TITLE
Added Ansible UTM Module for managing AAA groups.

### PR DESCRIPTION
**Pull request by maunzCache**
##### SUMMARY
This PR will add a new utm module which allows the management of aaa groups of the Sophos UTM. This module integrates with the existing utm_utils .

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
- module/web_infrastructure/sophos_utm/utm_aaa_group.py

### CODE REVIEW

**Comment by MatrixCrawler** [comment]
LGTM

shipit

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+DOCUMENTATION = """
+---
+module: utm_aaa_group
+
+author:
+    - Johannes Brunswicker (@MatrixCrawler)
+
+short_description: create, update or destroy reverse_proxy frontend entry in Sophos UTM
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py]: I think this is wrong. Could you please check description of this again ? Thanks.

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+description:
+    - Create, update or destroy a reverse_proxy frontend entry in SOPHOS UTM.
+    - This module needs to have the REST Ability of the UTM to be activated.
+
+version_added: "2.8"
+
+options:
+    name:
+        description:
+          - The name of the object. Will be used to identify the entry
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py]:
```suggestion
          - The name of the object. Will be used to identify the entry.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        required: true
+    adirectory_groups :
+        description:
+          - list of adirectory group strings
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py]:
```suggestion
          - list of adirectory group strings.
```

Comment (Akasurde) [reply lib/ansible/modules/web_infrastructure/sophos_utm/utm_aaa_group.py]: Same for other description.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please add Pytest unit test for this change - [here](https://github.com/ansible/ansible/tree/devel/test/units/modules/web_infrastructure)?

**Comment by maunzCache** [comment]
@Akasurde What do you want me to test? The module calls the utm_utils module and behaves like a AnsibleModule that fails with fail_json in case of an exception.
That means the exception case should be covered by the AnsibleModule class and the UTMModule call will fail if the utm_module code is changed.

I have no idea what or how I would test the current code :/

**Comment by maunzCache** [comment]
Just noticed my PR is a duplicate to https://github.com/ansible/ansible/issues/48926

**Comment by maunzCache** [comment]
@Akasurde Still need your feedback. @MatrixCrawler any idea regarding the unit test?

**Review by Akasurde — APPROVED** [review]
LGTM

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thanks all, merged

**Pull request merged**